src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComponentModelExtensions.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// These extension methods are used to mimic behavior from the .NET Framework where System.Attribute
+    /// had extra methods. For compatibility reasons, they are implemented with interfaces on the attributes
+    /// that need them, and then exposed via extension methods off of System.Attribute. The only difference
+    /// is that Attribute.TypeId is now accessed as Attribute.GetTypeId() as extension properties don't exist.
+    /// </summary>
+    internal static class ComponentModelExtensions
+    {
+        /// <summary>
+        /// This work around attempts to mimic the known cases where IsDefaultAttribute is true for System.ComponentModel
+        /// attributes. This does not account for derived attributes, which a few may be (most are sealed). The derived
+        /// attributes will most likely not be default as they will be overriding the default properties anyway, so this
+        /// heuristic will cover the majority of cases while allowing correct layering with System.ComponentModel.Primitives
+        /// </summary>
+        public static bool IsDefaultAttribute(this Attribute attribute)
+        {
+            Func<Attribute, bool> isDefaultAttribute;
+            if (s_defaultAttributes.TryGetValue(attribute.GetType(), out isDefaultAttribute))
+            {
+                return isDefaultAttribute(attribute);
+            }
+
+            return false;
```

Comment:
This could be done as:
```C#
Func<Attribute, bool> isDefaultAttribute;
return s_defaultAttributes.TryGetValue(attribute.GetType(), out isDefaultAttribute) && isDefaultAttribute(attribute);
```